box-share-sdk/src/test/java/com/box/androidsdk/share/sharerepo/ShareRepoTest.kt

```diff
@@ -0,0 +1,93 @@
+    private val mockFetchRolesResponse: BoxResponse<BoxCollaborationItem> = mock()
+    private val mockAddCollabsResponseTask: BoxFutureTask<BoxResponseBatch> = mock()
+    private val mockAddCollabsResponse: BoxResponse<BoxResponseBatch> = mock()
+
+    private lateinit var shareRepo: ShareRepo
+
+    @Before
+    fun setup() {
+        whenever(shareController.getInvitees(mockmShareItem, mockFilter)).thenReturn(mockGetInviteeResponseTask)
+        whenever(shareController.fetchRoles(mockmShareItem)).thenReturn(mockFetchRolesResponseTask)
+        whenever(shareController.addCollaborations(mockmShareItem, mockSelectedRole, mockEmailList)).thenReturn(mockAddCollabsResponseTask)
+        shareRepo = ShareRepo(shareController)
+        createStubs()
+    }
+
+    /**
+     * Mock callback responses.
+     */
+    private fun createStubs() {
+        doAnswer {
+            val callback = it.arguments[0] as BoxFutureTask.OnCompletedListener<BoxIteratorInvitees>
+            callback.onCompleted(mockGetInviteeResponse)
+            null
+        }.whenever(mockGetInviteeResponseTask).addOnCompletedListener(any<BoxFutureTask.OnCompletedListener<BoxIteratorInvitees>>())
+        doAnswer {
+            val callback = it.arguments[0] as BoxFutureTask.OnCompletedListener<BoxCollaborationItem>
+            callback.onCompleted(mockFetchRolesResponse)
+            null
+        }.whenever(mockFetchRolesResponseTask).addOnCompletedListener(any<BoxFutureTask.OnCompletedListener<BoxCollaborationItem>>())
+
+        doAnswer {
+            val callback = it.arguments[0] as BoxFutureTask.OnCompletedListener<BoxResponseBatch>
+            callback.onCompleted(mockAddCollabsResponse)
+            null
+        }.whenever(mockAddCollabsResponseTask).addOnCompletedListener(any<BoxFutureTask.OnCompletedListener<BoxResponseBatch>>())
+    }
+
+
+    @Test
+    fun `test fetch roles update LiveData values correctly`() {
+        assertNull(shareRepo.getFetchRoleItem().value) //initially the LiveData should not have any value
+        shareRepo.fetchRolesApi(mockmShareItem) //get a value and update as needed
+        assertEquals(mockFetchRolesResponse, shareRepo.getFetchRoleItem().value)
+    }
+
+
+    @Test
+    fun `test get invitees update LiveData values correctly`() {
+        assertNull(shareRepo.getInvitees().value) //initially the LiveData should not have any value
+        shareRepo.getInviteesApi(mockmShareItem, mockFilter) //get a value and update as needed
+        assertEquals(mockGetInviteeResponse, shareRepo.getInvitees().value)
+    }
+
+    @Test
+    fun `test get collab update LiveData values correctly` () {
+        assertNull(shareRepo.getAddCollabsBatch().value) //initially the LiveData should not have any value
+        shareRepo.addCollabsApi(mockmShareItem, mockSelectedRole, mockEmailList) //get a value and update as needed
+        assertEquals(mockAddCollabsResponse, shareRepo.getAddCollabsBatch().value)
+    }
```

Comment:
Any tests for failed cases?

Reply:
We will always get a response though. We only care about whether or not the liveData is updated to the response received. So I think this level of depth for this class is okay.